### ISSUE TITLE
System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription for stable servicing releases includes the full build information in the version suffix

### ISSUE THREAD

**Issue by DamianEdwards** [dotnet/runtime#45812]
**Description**
Using the just released 5.0.1 runtime, the `System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription` API is returning the string ".NET 5.0.1-servicing.20575.16". The issue reproduces on Windows during local dev on a framework-dependent app, and after deployment when running on Linux as a self-contained app.

**Repro**
Call the `System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription` API in an app targeting `net5.0` on a machine with the .NET 5.0.1 runtime installed.

**Expected**
Calling the `System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription` API should return a description with stable versions for stable servicing releases, e.g. ".NET 5.0.1".

**Actual**
Using the just released 5.0.1 runtime, calling the `System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription` API is returning the string ".NET 5.0.1-servicing.20575.16", which includes an unstable/preview version suffix.

**Comment by jkotas** [comment]
@Anipik @safern This API returns the product version string of System.Private.CoreLib. Is it expected that the product version string for 5.0.1 includes the unstable/preview version suffix?

**Comment by safern** [comment]
It shouldn't. I can take a look tomorrow to see why the version was wrong.

**Comment by DamianEdwards** [comment]
Is this fixed now?

**Comment by safern** [comment]
Yeah this was fixed on: https://github.com/dotnet/runtime/pull/45876. However I need to add tests for it to avoid regressing again.

**Comment by deeprobin** [comment]
@safern What is the current status of this?

**Comment by safern** [comment]
We are still missing tests for this.

**Comment by DamianEdwards** [comment]
This seems to not be fixed as I'm getting the following string on the .NET 6.0 GA build:
![image](https://user-images.githubusercontent.com/249088/140416216-4801bd12-a28c-406e-b989-97bb2e4e5ee0.png)

Output of `dotnet --info`;
```
.NET SDK (reflecting any global.json):
 Version:   6.0.100
 Commit:    9e8b04bbff

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22000
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.100\

Host (useful for support):
  Version: 6.0.0
  Commit:  4822e3c3aa

.NET SDKs installed:
  6.0.100-rc.2.21505.57 [C:\Program Files\dotnet\sdk]
  6.0.100 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.21 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.0-rc.2.21480.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.21 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.0-rc.2.21480.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.21 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.0-rc.2.21501.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by DamianEdwards** [comment]
Seems the `AssemblyInformationalVersion` in `System.Private.CoreLib` still has the full build info:

![image](https://user-images.githubusercontent.com/249088/140418127-f3a3a5e0-6483-4444-b1e4-93f3c5cad01c.png)

**Comment by DamianEdwards** [comment]
FYI seems we meant to take this for GA but it slipped through the cracks https://github.com/dotnet/runtime/pull/60572#issuecomment-948889384

**Comment by mairaw** [comment]
So is the fix going to make into 6.0.1? Just wondering if I need to have a workaround in place so we don't show an ugly version on the dot.net website as I reported on #61394. If it's just a few more days, I can wait.

**Comment by jeffhandley** [comment]
> So is the fix going to make into 6.0.1? Just wondering if I need to have a workaround in place so we don't show an ugly version on the dot.net website as I reported on #61394. If it's just a few more days, I can wait.

_7 months later..._

We don't have the work for this scheduled yet. Moving to Future. I don't know if a workaround is already in place though.
